galaxy/api/v2/views/collection.py

```diff
@@ -110,7 +110,9 @@ def _check_version_conflict(self, namespace, filename):
         except dj_exc.ObjectDoesNotExist:
             pass
         else:
-            raise CollectionExistsError()
+            raise CollectionExistsError(
+                f'Collection "{filename.namespace}-{filename.name}'
+                f'-{filename.version}" already exists.')
```

Comment:
Should we put this extra detail in the `_save_artifact()` `CollectionExistsError()` also?